### PR TITLE
DATA 498 add labels to geometries in RDK

### PR DESCRIPTION
**Pull request by bazile-clyde**
This commit is the first part of [DATA-498](https://viam.atlassian.net/browse/DATA-498). The next part will focus on the client and server interface. This seemed like a good stopping point since it's pretty large and self-contained.

### CODE REVIEW

File: components/camera/videosource/join_pointcloud.go [hunk 1]
```diff
@@ -218,7 +218,7 @@ func (jpcs *joinPointCloudSource) NextPointCloudNaive(ctx context.Context) (poin
 					savedDualQuat := spatialmath.NewZeroPose()
 					pcSrc.Iterate(numLoops, loop, func(p r3.Vector, d pointcloud.Data) bool {
 						if jpcs.sourceNames[iCopy] != jpcs.targetName {
-							spatialmath.ResetPoseDQTransalation(savedDualQuat, p)
+							spatialmath.ResetPoseDQTranslation(savedDualQuat, p)
```
Comment (bazile-clyde) [inline components/camera/videosource/join_pointcloud.go:221]: nit: typo in "Transalation"

File: pointcloud/point.go [hunk 2]
```diff
@@ -62,7 +62,7 @@ type Data interface {
 	// Note(erd): we should try to remove this in favor of immutability.
 	SetValue(v int) Data
 
-	// Value returns the intesity value, or 0 if it doesn't exist
+	// Intensity returns the intensity value, or 0 if it doesn't exist
```
Comment (bazile-clyde) [inline pointcloud/point.go:65]: nit: The function name is "Intensity" not "Value".

File: pointcloud/pointcloud_utils_test.go [hunk 3]
```diff
@@ -49,12 +49,13 @@ func TestBoundingBoxFromPointCloud(t *testing.T) {
 	}
 
 	for _, c := range cases {
-		expectedBox, err := spatialmath.NewBox(spatialmath.NewPoseFromPoint(c.expectedCenter), c.expectedDims)
+		expectedBox, err := spatialmath.NewBox(spatialmath.NewPoseFromPoint(c.expectedCenter), c.expectedDims, "")
 		test.That(t, err, test.ShouldBeNil)
-		box, err := BoundingBoxFromPointCloud(c.pc)
+		box, err := BoundingBoxFromPointCloudWithLabel(c.pc, "box")
```
Comment (bazile-clyde) [inline pointcloud/pointcloud_utils_test.go:54]: Updated this test to test the new `BoundingBoxFromPointCloudWithLabel` function, which is actually the old function's logic with a new function signature (it takes in a label now). It seemed liked the simplest way to cover the new function, `BoundingBoxFromPointCloudWithLabel`. See GitHub comment above this function for more info.

File: spatialmath/box.go [hunk 4]
```diff
@@ -118,7 +126,6 @@ func (b *box) CollidesWith(g Geometry) (bool, error) {
 	return true, newCollisionTypeUnsupportedError(b, g)
 }
 
-// CollidesWith checks if the given box collides with the given geometry and returns true if it does.
```
Comment (bazile-clyde) [inline spatialmath/box.go:121]: nit: This comment is a duplicate of the comment above and isn't actually for the `DistanceFrom` function; it's for the `CollidesWith` function.

File: spatialmath/geometry.go [hunk 5]
```diff
@@ -95,35 +101,35 @@ func (config *GeometryConfig) ParseConfig() (GeometryCreator, error) {
...
+		if creator, err := NewSphereCreator(config.R, offset, config.Label); err == nil {
 			return creator, nil
 		}
 		// never try to infer point geometry if nothing is specified
 	}
 	return nil, fmt.Errorf("%w %s", ErrGeometryTypeUnsupported, string(config.Type))
 }
 
-// NewGeometryFromProto instatiates a new Geometry from a protobuf Geometry message.
+// NewGeometryFromProto instantiates a new Geometry from a protobuf Geometry message.
```
Comment (bazile-clyde) [inline spatialmath/geometry.go:122]: nit: typo

**Comment by bhaney** [comment]
@raybjork Hey Ray, since you worked on geometries, we are adding a label field to Geometries, so that people can select on Geometries by user-defined names as well.

File: spatialmath/box.go [hunk 6]
```diff
@@ -14,26 +14,28 @@ import (
...
+func NewBoxCreator(dims r3.Vector, offset Pose, label string) (GeometryCreator, error) {
 	if dims.X <= 0 || dims.Y <= 0 || dims.Z <= 0 {
 		return nil, newBadGeometryDimensionsError(&box{})
 	}
-	return &boxCreator{dims.Mul(0.5), pointCreator{offset}}, nil
+	return &boxCreator{dims.Mul(0.5), pointCreator{offset, label}, label}, nil
 }
 
 // NewGeometry instantiates a new box from a BoxCreator class.
 func (bc *boxCreator) NewGeometry(pose Pose) Geometry {
```
Comment (bhaney) [inline spatialmath/box.go:37]: @raybjork  Do you think the new label argument go in the signature of BoxCreator, or in the signature of NewGeometry?

Comment (bhaney) [reply spatialmath/box.go:37]: Yea, I wondering if it's *only* NewGeometry() that should pass the label to the box{}/sphere{}/etc struct, or if it should be xCreator as you currently have it. I'll want to wait to hear from @raybjork to see how they use these creator, and if it makes sense to set the label at the creator level.

**Review by bhaney — APPROVED** [review]
I say this LGTM, but wait for Ray to comment on whether if boxCreator et al. should have the label field, or if labels shouldn't apply to xCreator structs and only should be within the Geometry structs themselves.

File: spatialmath/box.go (hunk 6 above)
Comment (biotinker) [reply spatialmath/box.go:37]: The GeometryCreator is the one that should have the label and provide it.

The way this information is used, is that any individual geometry is intended to be ephemeral and only valid at that location. The GeometryCreator persists the information about things like the dimensions of the geometry, and instantiates new versions of that geometry at various poses as requested.

There shouldn't be a situation where people are calling NewGeometry on a GeometryCreator and wanting it to have a different label, that should be an entirely new GeometryCreator.

Comment (bhaney) [reply spatialmath/box.go:37]: Excellent, then all the open questions are resolved on my end- LGTM @bazile-clyde

**Pull request merged**